egs/librispeech/ASR/conformer_ctc/conv1d_abs_attention.py

```diff
@@ -153,6 +153,19 @@ def __setstate__(self, state):
         if not hasattr(self, "padding_mode"):
             self.padding_mode = "zeros"
 
+import torch
+import torch.nn as nn
+m = nn.Tanh()
+
+def padding(input, padding_length):
+    # input shape : (B, T, D)
+    device = input.device
+    B, T, D = input.shape
+    src = torch.ones(B, T + 2*padding_length[0], D).to(device)
+    src[:, padding_length[0]:T+padding_length[0], :] = input
```

Comment:
I would recommend you to use [torch.nn.functional.pad](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html)
with `mode=constant` and `value=1`.

There is no need to initialize the allocated matrix with all ones.